### ISSUE TITLE
THE GUIS ARE TERRIBLE AND YOU SHOULD FEEL BAD

### ISSUE THREAD

**Issue by Funwayguy** [Funwayguy/BetterQuesting#163]
There, I made a thread so I can remind myself every day how much people dislike it. Now excuse me while I go cry in a corner...

(I am designing new concepts to replace it. Suggested ideas are welcome)

**Comment by drz2k** [comment]
Oh yeah you should feel bad! (Hope this is what you were expecting from us!)

Btw. the strange behaviour (closed issue) happened again! Terrible GUI! :D

Dunno, overall I like it. Only thing a bit annoying so far when mass creating quests is the "List","Item","Ok","Change Item","Add","Item","OK" clicky clicky ;)

**Comment by chrsgddrd** [comment]
What's needed to make a gui for your mod? How can I help? Have a few ideas and would love to throw my two cents in. Can I make a mock-up and you just dissect that? Or do you need the individual elements?

**Comment by Flabort** [comment]
I do agree that picking an item is a bit tedious. It would be nice if you could pick it in a few less clicks.

Also, the metadata selection bit is rough. I'm not sure what could be done to improve it, but I would like to be able to match part of the meta data without having to match it all (So for example I could check if they added Hammered to a tinkers pickaxe without having to have an exact duplicate of the pickaxe I had made).

**Comment by Funwayguy** [comment]
For anyone interested, this is what is heavily influencing my ideas on the next iteration of the editor:

![Node Editor](http://www.blog.dumontstudios.com/wp-content/uploads/2013/10/Quest-Editor.png)

@Flabort for that you want Partial NBT enabled and then remove all the tags from the tool you're not looking for.

**Comment by Flabort** [comment]
Hey, so what's the deal with Item Submission Stations? I was watching ItsLewdicolo on twitch, and he plays single player by connecting to a private server. 
He's playing Forever Stranded, which uses this mod, and he tries to use the Station, but the station GUI shows his inventory, four arrows, and nothing else. He had no clue what to do, said there was no quests to select there and didn't try clicking any of the arrows. 
1) Is the GUI supposed to look like that?
2) If so, is it possible to get a more straightforward GUI for it so it's easier to understand?

**Comment by Funwayguy** [comment]
@Flabort Firstly if you're going to report an issue please make a new thread for it instead of hijacking a less relevant one. Secondly, ItsLewdicolo is welcome to ask me himself without need of a messenger. As for the issue, yes the GUI is supposed to be like that and if the no quests show up then there are simply no active ones that support the station's use at that time.

**Comment by AriinPHD** [comment]
Didn't see this until now. I just opened a new thread/issue to request a taller window frame. I've closed that issue!

Commenting here instead:
*THANK YOU* for opening an issue for this (yourself). I've been realy opposed to using BQ since I first tried it because of the (sorry!!!) awful UI and UX.

Just to pitch in a bit more constrictive manner while the thread is open:
I'm begging on my knees for a taller window frame.
The current frame is so narrow (low in height) that it's almost mindboggling. It becomes a hassle to navigate the old (current) GUI because of the requirement to click-and-drag and zoom-out to use the window every time you open it up. It gets old really fast. I'm sure you have a ton of ideas, but I hope you, at the very least, make it taller. <3 Much love.

Really looking forward to the new version! I'm not sure why HQM feels "taller". They look identical (in height) but it's easier to use HQM for god knows what reason... >.<

**Comment by Funwayguy** [comment]
The reason that HQM seems to be easier to navigate is part of UI UX design involving distance between buttons. HQM keeps things pretty compact where BQ tries to spread out everything across the whole screen space (because I'm dumb). Another reason is that BQ doesn't use compact, self explanatory, icon buttons like HQM does (BQ didn't originally support them and I haven't updated the design). I'm also piggy backing off the crappy default button design Minecraft uses (I was lazy) instead of designing proper, purpose built buttons for the GUIs.

TLDR: I suck at UI UX but I know where I screwed up. It will just take a while to fix.

Other notable problems:
- The scrolling quest line keeps resetting position so you have to move it every... damn... time.
- The editor could really do with less screens and consolidate everything together
- Getting to your current quest from the home screen takes too long
- The mod developer is really lazy and easily distracted
- It's near impossible to understand what the heck the advanced options do
- The name of the mod is terrible
- Make the general UIs not take up so much screen space (editors need the space for tools)

PS: I've talked with LordDusk about the design as he's way better at it than me. Also we're not competing as much as people think we are and he's a pretty chill dude.

EDIT: I'm being sarcastic about the self hate, don't take it seriously.

**Comment by AriinPHD** [comment]
@Funwayguy 
Except your exaggerated self-criticism (Insert "I suck at [...]" here and there) I do love everything about your update/comment, and I for one greatly appreciate that you really take the time to respond to comments when you're not required to. Thanks! Also another thanks for being so cool about the criticism.

I look forward to seeing BQ grow! From the sounds of it you could get some design input from LordDusk. Huh. 8)

Hope you'll stop talking down on your own abilities. It's not a healthy trait and it's fine; we all (already) know that nobody is perfect.

And hells yeah LordDusk is one chill dude.

**Comment by Funwayguy** [comment]
Oh I'm not talking myself down in a serious way. I was being sarcastic and playing off the usual hate I get from various threads. I'll add an edit at the bottom for future reference.

**Comment by RareBeast** [comment]
As someone who is only using the mod (not creating quests etc) in Project Ozone 2. the scrolling issue is perhaps the most irritating.   Also, each quest category is bigger than the max zoom out so I have to zoom out as well as drag with the mouse to get to the quests on the far right or bottom of the window.

Other than that I'm finding the experience pretty good - not quite as good as previous HQM experiences, but you are definitely getting there.

Thanks

**Comment by MCJediYoda** [comment]
ability to look up recipes while in the quest item selector.  It helps coordinate quests when I know what the recipe is without having to take screen shots and have them on my second monitor.  if that makes any sense...  I have no problem using the existing gui and can navigate through them rather quickly it just took me a while to figure out. I will be using this mod in future mod packs.

**Comment by Thenosis** [comment]
I've started using this in my pack. I wanted to give it a try since using it in Project Ozone 2. Here are some suggestions for ease of use, quality of life. I find the BQM extremely easy to use. 

- Include a template (if possible) in an example resource folder so pack creators can use that to make their own title card instead of having to rely on looking up someone else's card. I have yet to find an example anywhere online yet either. That or include a template or measurements on the wiki page. My own title card looks really stretched and horrible right now. This is what it looks like and I have no idea what I'm doing wrong with it.

![2017-04-20_18 09 20](https://cloud.githubusercontent.com/assets/18347294/25254647/9fc64696-25f4-11e7-8dc4-0a79e481f1a6.png)

- See MCJediYoda's comment above. If at all possible add the ability for us to see the recipe when clicking on an item from the book itself. I don't know if this is at all possible though due to the functionality also being removed from HQM in recent iterations. I haven't been able to get it to work since playing Project Ozone. (1 not 2)

-EDIT If possible, add the ability for previously completed quest objectives to complete without the quest actually being open. Having to create two of an item that's only being asked for once can cause some material stock problems.

Other than those two items I don't see anything else wrong with BQM other than what's already been mentioned above. I'm looking forward to seeing what comes in the future for this mod. Thanks for making it.

**Comment by ghost** [comment]
Something like this?
![mockup](https://cloud.githubusercontent.com/assets/22108177/25508968/1fc94178-2b7b-11e7-973a-e81c4139a0a9.png)
-

- title screen should only be displayed once or when the home button is clicked on 2nd screen

- info icon should popup questline story/description, should be auto displayed on first view of the questline and be toggleable with the info icon
- If the questline description is open and the user clicks on another questline, that questlines description should show

- navigation icons on 2nd screen can be above or below the questline scrollbox (above may be better to keep all the controls together and to keep near the first questlines)
- the x exit/close gui can be in the questline box or semi-transparent in top right of the quest tree pane, escape should also close the gui  
- fewer modal screens is better (probably want to never go more than three screens deep: Title -> Quests ->  Questline edit/Quest edit/etc)
- the quest or storyline info popup in my example could also be used as the quest editor as well (although using the 2nd screen as a template for a variety of complex actions may be better for quests or things such as categorized options).  
- In edit mode you could also add a + and - and pencil icon in the top/bottom left of the second pane to add/remove/edit questlines 
 
![mockup2](https://cloud.githubusercontent.com/assets/22108177/25509649/2c907c46-2b80-11e7-9020-5947a8c81ea4.png)


moqups.com if you want to mockup a layout (no need for an account, trial works just fine)

**Comment by Funwayguy** [comment]
There are a bunch of places where this would likely fail UI UX nor does it really solve the complex problem of advanced NBT/JSON based editors but I appreciate the effort.
